server/hotzone/models.py

```diff
@@ -9,3 +10,72 @@ def __str__(self):
     y_coord = models.FloatField('Y Co-ordinate')
     name = models.CharField(max_length=100)
     address = models.CharField(max_length=250, blank = True)
+
+''' Disease class '''
+class Virus(models.Model):
+
+	class Meta:
+		constraints = [
+			models.UniqueConstraint(fields = ['name'], name='unique-name')
+		]
+
+	name = models.CharField(max_length=64)
+	disease = models.CharField(max_length=150)
+	max_infectious = models.IntegerField() #no. of days
+	
+	def __str__(self):
+		return self.name
+
+''' Patient class '''
+class Patient(models.Model):
+
+	class Meta:
+		constraints = [
+			models.UniqueConstraint(fields = ['hkid'], name='unique-hkid')
+		]
+
+	name = models.CharField(max_length=150)
+	hkid = models.CharField(max_length=9)
+	dob = models.DateField()
+	
+	def __str__(self):
+		return self.hkid
+
+''' Case class '''
+class Case(models.Model):
+
+	class Meta:
+		constraints = [
+			models.UniqueConstraint(fields = ['case_no'], name='unique-case-no')
+		]
+
+	# enum type 
+	class Origin(models.TextChoices):
+		LOCAL = 'L'
+		IMPORTED = 'I'
+
+	case_no = models.IntegerField()
+	confirmed = models.DateField()
+	origin = models.CharField(max_length=7, choices=Origin.choices, default=Origin.LOCAL)
+	patient = models.ForeignKey(Patient, on_delete=models.CASCADE)	
+	infected = models.ForeignKey(Virus, on_delete=models.CASCADE)
```

Comment:
let's name this field "virus" for clarity

Reply:
Feel free to merge after changing this 👍

Reply:
Oh good catch

Reply:
ok line 61 `infected `-> `virus`